### PR TITLE
Allow mixing of Fold.Node/Tensor in calls to Fold.add

### PR DESCRIPTION
**Pull request by rshin**

### CODE REVIEW

File: pytorch_tools/torchfold.py [hunk 1]
```diff
@@ -111,34 +111,42 @@ def _batch_args(self, arg_lists, values):
...
+                                       for arg_item in arg], 0))
                 else:
-                    for i in range(2, len(arg)):
-                        if arg[i] != arg[0]:
-                            raise ValueError("Can not use more then one of nobatch argument, got: %s." % str(arg))
-                    x = arg[0]
-                    res.append(x.get(values))
-            elif isinstance(arg[0], (torch.tensor._TensorBase, Variable)):
-                res.append(torch.cat(arg, 0))
+                    assert all(arg[0] == arg_item for arg_item in arg[1:])
```
Comment (ilblackdragon) [inline pytorch_tools/torchfold.py]: let's keep ValueError here with error message "Can not use more then one of nobatch argument, got: %s." % str(arg) so it's clear what happened.

File: pytorch_tools/torchfold.py [hunk 2]
```diff
@@ -111,34 +111,42 @@ def _batch_args(self, arg_lists, values):
...
+                    elif isinstance(arg_item, (torch.tensor._TensorBase,
+                        Variable)):
+                        r.append(arg_item)
                     else:
-                        var = Variable(torch.LongTensor(arg), volatile=self.volatile)
-                    res.append(var)
-                except:
-                    print("Constructing LongTensor from %s" % str(arg))
-                    raise
+                        raise Exception(
```
Comment (ilblackdragon) [inline pytorch_tools/torchfold.py]: ValueError to make exception more specific?

**Review by ilblackdragon — APPROVED** [review]
Few error handling changes, otherwise LGTM

File: pytorch_tools/torchfold.py (hunk 1 above)
Comment (rshin) [reply pytorch_tools/torchfold.py]: Done

Comment (rshin) [reply pytorch_tools/torchfold.py]: Done

File: pytorch_tools/torchfold.py (hunk 2 above)
Comment (rshin) [reply pytorch_tools/torchfold.py]: Done

**Pull request merged**